lib/mix/lib/mix/local/installer.ex

```diff
@@ -29,33 +49,37 @@ defmodule Mix.Local.Installer do
   def install({module, name}, argv, switches) do
     {opts, args} = OptionParser.parse!(argv, strict: switches)
 
-    case args do
-      [src] ->
-        with :ok <- check_argument(src), :ok <- module.check_path_or_url(src) do
-          do_install({module, name}, src, opts)
-        else
-          {:error, message} -> Mix.raise message <> "\n" <> usage(name)
+    install_spec = parse_args(args, opts)
+
+    case module.check_install_spec(install_spec, opts) do
+      :ok -> :noop
+      {:error, message} -> Mix.raise message <> "\n" <> usage(name)
```

Comment:
Btw, I think we should revisit `parse_args` and append `"\n" <> usage(name)` to many of its error messages. We should revisit the `usage/1` function itself as well.